### PR TITLE
Update thunder from 4.0.1.14502 to 4.0.3.31366

### PR DESCRIPTION
**Pull request by ElfSundae**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by bevanjkay** [comment]
@cho-m The livecheck block uses the URL provided by @ElfSundae in #107460 
I'm not certain if it would need the second step to fetch the appcast, or whether pulling the version from the appcast URL would suffice. Also need to consider the reliability of the initial URL used - I have had some curl errors in testing.

**Review by miccal — DISMISSED** [review]

**Comment by cho-m** [comment]
> I'm not certain if it would need the second step to fetch the appcast, or whether pulling the version from the appcast URL would suffice.

If the upstream continues using a version-specific appcast URL for releases, then it should be fine to simplify to just a `regex(...)` stanza with the first capture pattern.

I don't have any data to check so it is hard to say for sure.

For now, we can try simplifying to only the first URL as it avoids the overhead of an extra HTTP request.

---

> Also need to consider the reliability of the initial URL used - I have had some curl errors in testing, it was previously used as the appcast URL before the livecheck was first added.

I also had some trouble with URL, but I have had trouble with some other Chinese Casks' URLs too.

I think that as long as it works with Homebrew's CI and it works for the users who will most likely help update the Cask (e.g. `homebrew-cask`'s user base in China), I think it should be fine to use it.

If anything goes wrong, we can easily revert it.

File: Casks/thunder.rb [hunk 1]
```diff
@@ -10,9 +10,12 @@
...
   livecheck do
-    url :homepage
-    strategy :page_match
-    regex(/thunder_(\d+(?:\.\d+)*)\.dmg/i)
+    url "https://static-xl9-ssl.xunlei.com/json/mac_preferences.json"
+    strategy :page_match do |page|
+      version = page[/appcast_(\d+(?:\.\d+)*)\.xml/i, 1]
+      version_page = Homebrew::Livecheck::Strategy.page_content("https://down.sandai.net/mac/thunder/appcast_#{version}.xml")
+      version_page[:content].scan(/thunder[_-](\d+(?:\.\d+)*)\.dmg/i).flatten
+    end
```
Comment (cho-m) [inline Casks/thunder.rb:18]: As mentioned in previous comment, let's try using simpler livecheck for now. If it breaks, we can add extra logic to deal with it:
```suggestion
    regex(/appcast[._-]v?(\d+(?:\.\d+)+)\.xml/i)
```

Also some regex-style changes to align with Formula-side (e.g. https://docs.brew.sh/Formula-Cookbook#livecheck-blocks) like using more lenient `[._-]` for in-between characters, etc.

**Comment by bevanjkay** [comment]
@cho-m Well, we're seeing the curl error already. In my testing it seemed to be a success rate of about 50%.

**Comment by bevanjkay** [comment]
@cho-m After a requeue this has now passed CI. From testing even the homepage was returning curl errors sporadically, so I don't think this change should reduce the reliability.

**Review by cho-m — APPROVED** [review]

**Pull request merged**